### PR TITLE
Quorum queues: chunk consumer log effects

### PR DESCRIPTION
**Pull request by kjnilsson**
Chunk quorum queue deliveries

This puts a limit to the amount of message data that is added
to the process heap at the same time to around 128KB.

Large prefetch values combined with large messages could cause
excessive garbage collection work.

Also simplify the intermediate delivery message format to avoid
allocations that aren't necessary.

### CODE REVIEW

File: deps/rabbit/src/rabbit_fifo.erl [hunk 1]
```diff
@@ -1978,18 +2000,20 @@ get_next_msg(#?MODULE{returns = Returns0,
             Msg
     end.
 
-delivery_effect({CTag, CPid}, [{Idx, {MsgId, Header}}],
+delivery_effect({CTag, CPid}, [{MsgId, [Idx | Header]}],
```
Comment (ansd) [inline deps/rabbit/src/rabbit_fifo.erl:2003]: Use `?MSG(Idx, Header)` instead of `[Idx | Header]` ?

File: deps/rabbit/src/rabbit_fifo.erl [hunk 2]
```diff
@@ -1942,12 +1942,34 @@ evaluate_limit(Index, Result, BeforeState,
...
+                [CurChunk | Chunks])
+  when Bytes >= ?DELIVERY_CHUNK_LIMIT_B ->
+    Size = get_header(size, Header),
+    chunk_disk_msgs(Rem, Size, [[], [Msg | CurChunk] | Chunks]);
+chunk_disk_msgs([{_MsgId, ?MSG(_RaftIdx, Header)} = Msg | Rem], Bytes,
+                [CurChunk | Chunks]) ->
+    Size = get_header(size, Header),
+    chunk_disk_msgs(Rem, Bytes + Size, [[Msg | CurChunk] | Chunks]).
+
+append_delivery_effects(Effects0, AccMap, _State)
```
Comment (ansd) [inline deps/rabbit/src/rabbit_fifo.erl:1962]: Not part of this PR, but I guess, we should rename `append_delivery_effects` to `prepend_delivery_effects` since the delivery effects are prepended, right?

File: deps/rabbit/src/rabbit_fifo.erl (hunk 1 above)
Comment (kjnilsson) [reply deps/rabbit/src/rabbit_fifo.erl]: cheers done

**Review by ansd — APPROVED** [review]

**Comment by michaelklishin** [comment]
It was decided to ship this only in `3.12.0` and not to backport.
